src/custom/hooks/useSwapCallback.ts

```diff
@@ -31,9 +34,11 @@ export function useSwapCallback(
   const validTo = useTransactionDeadline()?.toNumber() || MAX_VALID_TO_EPOCH
   const addPendingOrder = useAddPendingOrder()
   const { INPUT: inputAmount, OUTPUT: outputAmount } = computeSlippageAdjustedAmounts(trade, allowedSlippage)
+  const weth = useWETHContract()
+  const wrapEther = useWrapEther()
 
   return useMemo(() => {
-    if (!trade || !library || !account || !chainId || !inputAmount || !outputAmount) {
+    if (!trade || !library || !account || !chainId || !inputAmount || !outputAmount || !weth) {
```

Comment:
I guess that here, u would need to check that wrapEther is not null (I understand returns null if WETH is not available...) Related to my other comment.